autoblocks/_impl/global_state.py

```diff
@@ -25,10 +29,19 @@ def init() -> None:
         daemon=True,
     )
     background_thread.start()
-
+    for sig in [signal.SIGINT, signal.SIGTERM]:
+        _loop.add_signal_handler(sig, lambda: asyncio.ensure_future(_on_exist_signal()))
     _started = True
 
 
+async def _on_exist_signal() -> None:
```

Comment:
also i think this function could use some comments, and maybe a link to the blog you followed